docs/DeveloperGuide.adoc

```diff
@@ -170,168 +216,116 @@ Classes used by multiple components are in the `tatracker.commons` package.
-
-image::CommitActivityDiagram.png[]
-
-==== Design Considerations
-
-===== Aspect: How undo & redo executes
-
-* **Alternative 1 (current choice):** Saves the entire TA-Tracker.
-** Pros: Easy to implement.
-** Cons: May have performance issues in terms of memory usage.
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the student being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
-
-===== Aspect: Data structure to support the undo/redo commands
-
-* **Alternative 1 (current choice):** Use a list to store the history of TA-Tracker states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedTaTracker`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+[[Student-View]]
+=== Student View
+*Student view* is the term used to characterise the different functionalities
+related to the students.
 
-//tag::moduleview[]
-=== Module View
-Module view is the term used to characterise the different functionalities
-related to the modules and groups that the user is affiliated with.
+Students are a part of *groups* and groups are a part of *modules*.
 
 ==== Model Framework
 The following class diagram shows how different classes are related in the
-functioning of the module view.
+functioning of the Student View.
 
 .Module View - Class Diagram
 image::ModuleModelClassDiagram.png[]
 
-The TaTracker model class contains a UniqueModuleList which helps it keep track
-of the different modules the user is associated with. Each module contains a
-UniqueGroupList and a UniqueSessionList.
+The TaTracker `Model` class contains a `UniqueModuleList` which helps it keep track
+of the different *modules* the user is associated with. Each `module` contains a
+`UniqueGroupList` and a `UniqueSessionList`.
+
+The `UniqueGroupList` contains a list of all the *groups* of a module that the user
+is affiliated with. Each group contains a `UniqueStudentsList` that contains the
+*students* in that group.
 
-The UniqueGroupList contains a list of all the groups of a module that the user
-is affiliated with. Each group contains a UniqueStudentsList that contains the
-students in that group.
+The `UniqueSessionList` contains a list of all the *sessions that have already been
+done* associated with the module. This list is used in the TSS view.
```

Comment:
Time for a diagram!

Reply:
Isn't there a class diagram there?

Reply:
There is a class diagram before the explanation that shoes all of this :)